tox.ini

```diff
@@ -1,13 +1,21 @@
 [tox]
-envlist = py27,py33,py34,py35,pypy
+envlist = py{27,34,35,36,37,py,py3}, flake8
+
+[testenv:flake8]
+basepython = python3.7
+deps = flake8>=3.7,<4
+commands =
+    flake8 graphql_relay tests
+
+[testenv:pypy]
+whitelist_externals=*
+
+[testenv:pypy3]
+whitelist_externals=*
 
 [testenv]
 deps=
-    pytest>=2.7.2
-    django>=1.8.0,<1.9
-    six
-    flake8
-    singledispatch
+    pytest>=4.6,<5
```

Comment:
Why not use the test deps here instead of defining it twice?

Reply:
Ok - I'm using "setup.py test" now, relying on "test_require" to specify the requirements. This is also how it is described in the README.

Note that this will be changed in v3 where we use pytest 5 which does not support "setup.py test" any more. But in v2 we're forced to stick with pytest 4 anyway because pytest 5 doesn't support Python 3, so I think it's ok to run tests the "old way" here.

Reply:
👍 makes sense thanks @Cito